core/compute/compute.go

```diff
@@ -166,11 +166,23 @@ func DoInstanceDeactivate(instance model.Instance, client *client.Client, timeou
 	if utils.IsNoOp(instance.ProcessData) {
 		return nil
 	}
-	t := time.Duration(timeout) * time.Second
+
 	container, err := utils.GetContainer(client, instance, false)
 	if err != nil {
 		return errors.Wrap(err, constants.DoInstanceDeactivateError+"failed to get container")
 	}
+
+	inspect, err := client.ContainerInspect(context.Background(), container.ID)
+	if err != nil {
+		return errors.Wrap(err, constants.GetInstanceHostMapDataError+"failed to inspect container")
+	}
+
+	if inspect.Config.StopTimeout != nil {
+		timeout = *inspect.Config.StopTimeout
```

Comment:
thinking if we should pick the max value between `stop_grace_period` and `timeout` set by rancher container.stop command